src/App.js

```diff
@@ -14,11 +17,99 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: [],
+  }
+
+  showGoodsList = () => {
+    this.setState({ goods: [...goodsFromServer] });
+  }
```

Comment:
It would be good to change the logic here a bit. It would be better to create in the state the value, for example, `areGoodsVisible`, and render 'show' button and `GoodsList` according to this value:

`{this.state.areGoodsVisible && <GoodsList />}`
and
`{!this.state.areGoodsVisible && <button>}`
When button is clicked you should change that value:

```
showGoodsList = () => {
  this.setState(state => {(
      areGoodsVisible: !state.areGoodsVisible
    )}
  )
}

```